lib/AesCm128HmacSha256EncryptionKey.js

```diff
@@ -184,14 +191,28 @@ export class AesCm128HmacSha256EncryptionKey
 		return [payload, authTag];
 	}
 	
-	async ratchet()
+	async ratchet(index)
 	{
+		// Keep track of how many ratchet keys we have
+		if (index > this.ratchetIndex) {
+			this.ratchetIndex = index;
+		}
```

Comment:
Seems counter-intuitive, IMO.

Reply:
Actually I can delete all of this sorry.
I was testing out keeping track of ratchet in order to stop decryption when the limit was reached.